test/sipreg.c

```diff
@@ -168,6 +174,15 @@ static int reg_test(enum sip_transp tp, bool deprecated, uint16_t srcport)
 	TEST_ASSERT(srv->n_register_req > 0);
 	TEST_ASSERT(test.n_resp > 0);
 
+	if (!deprecated) {
+		contact_hdr = sip_msg_hdr(srv->sip_msgs[srv->n_register_req - 1],
+			SIP_HDR_CONTACT);
+		err = re_regex(contact_hdr->val.p, contact_hdr->val.l,
+			";" CPARAMS ">;expires=", &contact_match);
```

Comment:
Really wanted to add some test. I have the full matching regex for both deprecated or not, took me a while to figure out it's an interesting version of regex.... So I'd just settle on this instead of having the deprecated version testing some other stuff while this one just some substr.

Reply:
yes, please complete the test then we'll see if we need more.

Reply:
>  yes, please complete the test! Then we'll see if we need more.


The test already confirms if cparam is supplied it's added correctly, currently there's no way to add cparam through the deprecated interface.  IMHO, this should be a good enough test case.

What I meant above is that, originally I wanted to define the test to be "validate the whole Contact URI string", thus one less branch dependent test, just supply different expression based on the flag. Particularly since it's now possible through the new sip message capturing.

However since the regex implementation doesn't support wildcard, after quite a lot of effort I don't think it's worth it.

What I meant above is just that I'm effectively using regex to do a substr test, bit of a waste,  but then it's a test cases so..

Reply:
It's not used, just testing it's found. Just wanted to do better than `tmp` .. perhaps better differs upon beholder, in this case it confuses unfortunately. Character class was what I meant by a lot of effort.. we're talking about correctly testing the new interface + verifying it doesn't break existing behavior, which is done here; versus, introducing a new and possibly better test.

I actually could've done quite a lot with `[^;];[^;]`, but then they're variable based on actual test case callers so it's going to introduce limitation. If it were my code base it's a hard no that introduces more troubles and code than it solves, but i'm not the maintainer here so if you insist

Reply:
My initial attempt extended regex, before I discovering it's not standard regex impl, if you could suggest how to translate, happy to update:

```
err = re_regex(contact_hdr->val.p, contact_hdr->val.l, deprecated ?
                        "^<sip:[^>]+>;.*$" : "^<sip:([^;]+;)+some-param=test;other-param=123;>;.*$", ...)
TEST_ERR(err)
```

If we stick to just validating the actual new behavior, can swap out that `contact_match` with a NULL, I thought the segfault I got was from using NULL. Or just a simple substring test. 

Note that I've already gotten a sense of the level of test in the code base for similar intrusiveness then try to tune it up a step, and that resulted in the previous a bit.